### PR TITLE
Use snprintf instead of sprintf

### PR DESCRIPTION
**Pull request by hughbe**
No real functional changes here, just more safe

### CODE REVIEW

**Comment by hughbe** [comment]
@swift-ci please smoke test

File: lib/Commands/BuildEngineCommand.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ struct AckermannKey {
   /// Convert an Ackermann key to its encoded representation.
   operator core::KeyType() const {
     char inputKey[32];
-    sprintf(inputKey, "ack(%d,%d)", m, n);
+    snprintf(inputKey, sizeof(inputKey), "ack(%d,%d)", m, n);
```
Comment (neonichu) [inline lib/Commands/BuildEngineCommand.cpp]: We should use the statically known size here (and everywhere else) instead of `sizeof`.

Comment (ddunbar) [inline lib/Commands/BuildEngineCommand.cpp]: @neonichu I'm not sure what you mean, I think `sizeof` is appropriate for this use case. What else would you write?

**Comment by hughbe** [comment]
I have no real opinion either way, I realise a bunch of these are in unit tests anyways

**Pull request closed without merging**